frontend/packages/metal3-plugin/src/components/dashboard/inventory-card.tsx

```diff
@@ -0,0 +1,142 @@
+import { K8sResourceKind, referenceForModel } from '@console/internal/module/k8s';
+import { FirehoseResource } from '@console/internal/components/utils';
+import { MachineModel } from '@console/internal/models';
+import { getNamespace } from '@console/shared';
+
+import { getHostStorage, getHostNICs, getHostCPU, getHostMachineName } from '../../selectors';
+import { getInventoryQueries, HostQuery } from './queries';
+
+const getResources = (namespace: string, machineName: string): FirehoseResource[] => [
+  {
+    isList: false,
+    namespace,
+    name: machineName,
+    kind: referenceForModel(MachineModel),
+    prop: 'machine',
+  },
+];
+
+export const InventoryCard: React.FC<InventoryCardProps> = ({
+  obj,
+  watchPrometheus,
+  stopWatchPrometheusQuery,
+  prometheusResults,
+  watchK8sResource,
+  stopWatchK8sResource,
+  resources,
+}) => {
+  const namespace = getNamespace(obj);
+  const machineName = getHostMachineName(obj);
+  const k8sResources = getResources(namespace, machineName);
+  const machineAddresses = _.get(resources.machine, 'data.status.addresses', []);
+  const machineIP = _.get(machineAddresses.find((addr) => addr.type === 'InternalIP'), 'address');
+
+  React.useEffect(() => {
+    k8sResources.forEach((r) => watchK8sResource(r));
+    return () => {
+      k8sResources.forEach((r) => stopWatchK8sResource(r));
+    };
+  }, [watchK8sResource, stopWatchK8sResource, k8sResources]);
+
+  React.useEffect(() => {
+    if (machineIP) {
+      const queries = getInventoryQueries(machineIP);
+      Object.keys(queries).forEach((key) => watchPrometheus(queries[key]));
+      return () => {
+        Object.keys(queries).forEach((key) => stopWatchPrometheusQuery(queries[key]));
+      };
+    }
+    return undefined;
+  }, [watchPrometheus, stopWatchPrometheusQuery, machineIP]);
+
+  const diskCount = getHostStorage(obj).length;
+  const nicCount = getHostNICs(obj).length;
+  const cpuCount = getHostCPU(obj).count;
+
+  const queries = getInventoryQueries(machineIP);
+
+  const otherPodCount = _.get(
+    prometheusResults.getIn([queries[HostQuery.NUMBER_OF_PODS], 'result'], {}),
+    'data.result',
+    [],
+  ).length;
+
+  const fanResult = _.get(
+    prometheusResults.getIn([queries[HostQuery.NUMBER_OF_FANS], 'result'], {}),
+    'data.result',
+    [],
+  );
+  const fanCount = fanResult.length;
+  const fanNotOkCount = fanResult.filter((el) => el.metric.status !== 'ok').length;
+
+  const psuResult = _.get(
+    prometheusResults.getIn([queries[HostQuery.NUMBER_OF_PSUS], 'result'], {}),
+    'data.result',
+    [],
+  );
+  const psuCount = psuResult.length;
+  const psuNotOkCount = psuResult.filter((el) => el.metric.status !== 'ok').length;
+
+  return (
+    <DashboardCard>
+      <DashboardCardHeader>
+        <DashboardCardTitle>Inventory</DashboardCardTitle>
+      </DashboardCardHeader>
+      <DashboardCardBody>
+        <InventoryItem
+          isLoading={false}
```

Comment:
this should be loading until `prometheusResults.getIn([queries[HostQuery.NUMBER_OF_PODS], 'result']` actually returns something.

Same for other queries/loading stuses of items